test/modules/MetricsValidator/src/tests/ValidateHostRanges.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace MetricsValidator.Tests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Edge.Agent.Diagnostics;
+    using Microsoft.Extensions.Logging;
+
+    public class ValidateHostRanges : TestBase
+    {
+        readonly string endpoint = Guid.NewGuid().ToString();
+
+        public ValidateHostRanges(TestReporter testReporter, IMetricsScraper scraper, ModuleClient moduleClient)
+            : base(testReporter, scraper, moduleClient)
+        {
+        }
+
+        protected override async Task Test(CancellationToken cancellationToken)
+        {
+            List<Metric> metrics = (await this.scraper.ScrapeEndpointsAsync(cancellationToken)).ToList();
+            this.CheckCPU(metrics);
+        }
+
+        void CheckCPU(List<Metric> metrics)
+        {
+            const string cpuMetricName = "edgeAgent_used_cpu_percent";
+            var cpuMetrics = metrics.Where(m => m.Name == cpuMetricName);
+            this.testReporter.Assert($"{cpuMetricName} metric exists", cpuMetrics.Any(), $"Missing {cpuMetricName}");
+
+            var hostCpu = cpuMetrics.Where(m => m.Tags.TryGetValue("module", out string module) && module == "host").ToDictionary(m => m.Tags["quantile"], m => m.Value);
+            this.testReporter.Assert("Host has all quantiles", hostCpu.Count == 6, $"Host had the following quantiles: {string.Join(", ", hostCpu.Keys)}");
```

Comment:
What is `hostCpu.Count == 6` doing here?

Reply:
Our histograms return 6 quantiles (50, 90, 95, 99, 99.9, 99.99)

Reply:
I'll add a comment